scripts/pylib/twister/twisterlib/runner.py

```diff
@@ -674,22 +674,25 @@ def process(self, pipeline, done, message, lock, results):
         # Run the generated binary using one of the supported handlers
         elif op == "run":
             logger.debug("run test: %s" % self.instance.name)
-            self.run()
-            logger.debug(f"run status: {self.instance.name} {self.instance.status}")
             try:
+                self.run()
+                logger.debug(f"run status: {self.instance.name} {self.instance.status}")
+
                 # to make it work with pickle
                 self.instance.handler.thread = None
                 self.instance.handler.duts = None
-                pipeline.put({
-                    "op": "report",
-                    "test": self.instance,
-                    "status": self.instance.status,
-                    "reason": self.instance.reason
-                    }
-                )
-            except RuntimeError as e:
-                logger.error(f"RuntimeError: {e}")
-                traceback.print_exc()
+            except Exception as e:
+                logger.error(f"Error: {e}")
+                self.instance.status = "error"
+                self.instance.reason = "general runtime exception"
+
```

Comment:
```suggestion
                d_log = f"{self.instance.build_dir}/device.log"         
                with open(d_log, "a") as dlog_fp:
                    dlog_fp.write(str(e))

```
without it reports contain useless build.log.